src/target/renesas_rz.c

```diff
@@ -202,6 +211,21 @@ static const char *renesas_rz_part_name(const uint32_t part_id)
 	return "Unknown";
 }
 
+/*
+ * Function to read RAM incrementally to determine the size of it for the given model.
+ */
+static uint32_t renesas_rz_ram_size(target_s *target)
+{
+	uint32_t result = 0;
```

Comment:
Given this is only used in the loop and is otherwise not read, perhaps move this to just inside the loop and drop the init? Should improve codegen a touch and improves scoping.